### PR TITLE
New modules: agat/spstatistics and agat/sqstatbasic Issue #2257

### PR DESCRIPTION
**Pull request by toniher**
Added new module. agat/statistics - Reference: https://github.com/nf-core/modules/issues/2257
Two modes that refer to two scripts in AGAT package.
* normal
* basic

### CODE REVIEW

File: modules/nf-core/agat/statistics/main.nf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    tuple val(meta), path("*.txt"), emit: stats_file
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def program = args.contains("--basic") ? "agat_sq_stat_basic.pl" : "agat_sp_statistics.pl"
```
Comment (JoseEspinosa) [inline modules/nf-core/agat/statistics/main.nf:23]: I would say that the guidelines here will be to have a different module for each of the commands, something such as: `agat/spstatistics` and `agat/sqstatbasic`

Comment (JoseEspinosa) [reply modules/nf-core/agat/statistics/main.nf:23]: Also `args` is used to pass the tool arguments itself, in this case `-i` or `--gff`

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
I think this should be split into two different modules. Also, you need to modify the [`pytest.yml`](https://github.com/JoseEspinosa/nf-core-modules/blob/cace92a17e50a4b324e59f29cb5a830a38de70b3/tests/config/pytest_modules.yml) file so that the tests are run for the module, maybe you missed to commit it if you used nf-core tools to create the module

**Comment by toniher** [comment]
@nf-core-bot fix linting

File: modules/nf-core/agat/statistics/main.nf (hunk 1 above)
Comment (toniher) [reply modules/nf-core/agat/statistics/main.nf]: Changes applied. Split in two modules. Included missing ```pytest_modules.yml``` additions.

File: modules/nf-core/agat/spstatistics/main.nf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+
```
Comment (JoseEspinosa) [inline modules/nf-core/agat/spstatistics/main.nf:25]:
```suggestion
```

File: modules/nf-core/agat/spstatistics/main.nf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    conda (params.enable_conda ? "bioconda::agat=1.0.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/agat:1.0.0--pl5321hdfd78af_0' :
+        'quay.io/biocontainers/agat:1.0.0--pl5321hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(gff)
+
+    output:
+    tuple val(meta), path("*.txt"), emit: stats_file
```
Comment (JoseEspinosa) [inline modules/nf-core/agat/spstatistics/main.nf:14]:
```suggestion
    tuple val(meta), path("*.txt"), emit: stats_txt
```

File: modules/nf-core/agat/spstatistics/meta.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  - statistics
+
+tools:
+  - agat:
+      description: "AGAT is a toolkit for manipulation and getting information from GFF/GTF files"
+      homepage: "https://github.com/NBISweden/AGAT"
+      documentation: "https://agat.readthedocs.io/"
+      tool_dev_url: "https://github.com/NBISweden/AGAT"
+      doi: "10.5281/zenodo.3552717"
+      licence: "['GPL v3']"
```
Comment (JoseEspinosa) [inline modules/nf-core/agat/spstatistics/meta.yml:17]:
```suggestion
      licence: ["GPL v3"]
```

File: modules/nf-core/agat/sqstatbasic/main.nf [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    conda (params.enable_conda ? "bioconda::agat=1.0.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/agat:1.0.0--pl5321hdfd78af_0' :
+        'quay.io/biocontainers/agat:1.0.0--pl5321hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(gff)
+
+    output:
+    tuple val(meta), path("*.txt"), emit: stats_file
```
Comment (JoseEspinosa) [inline modules/nf-core/agat/sqstatbasic/main.nf:14]:
```suggestion
    tuple val(meta), path("*.txt"), emit: stats_txt
```

File: modules/nf-core/agat/sqstatbasic/main.nf [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+
```
Comment (JoseEspinosa) [inline modules/nf-core/agat/sqstatbasic/main.nf:25]:
```suggestion
```

File: modules/nf-core/agat/sqstatbasic/meta.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  - statistics
+
+tools:
+  - agat:
+      description: "AGAT is a toolkit for manipulation and getting information from GFF/GTF files"
+      homepage: "https://github.com/NBISweden/AGAT"
+      documentation: "https://agat.readthedocs.io/"
+      tool_dev_url: "https://github.com/NBISweden/AGAT"
+      doi: "10.5281/zenodo.3552717"
+      licence: "['GPL v3']"
```
Comment (JoseEspinosa) [inline modules/nf-core/agat/sqstatbasic/meta.yml:17]:
```suggestion
      licence: ["GPL v3"]
```

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Just some small suggestions, otherwise looks awesome! :smile:

**Review by JoseEspinosa — APPROVED** [review]
Thanks @toniher !

**Pull request merged**